### ISSUE TITLE
PreferJavaTimeOverload false positive for Duration#dividedBy(long)

### ISSUE THREAD

**Issue by michaelhixson** [google/error-prone#1431]
`PreferJavaTimeOverload` reports an incorrect warning when it sees `Duration#dividedBy(long)`.

Example code:
```java
Duration totalTime = Duration.ofSeconds(10);
long numberOfPeople = 5; // does not represent a Duration
Duration timePerPerson = totalTime.dividedBy(numberOfPeople);
```

Expected behavior: No warnings.

Actual behavior:
> [PreferJavaTimeOverload] If the numeric primitive (numberOfPeople) represents a Duration, please call totalTime.dividedBy(Duration) instead.

I'm using Error Prone 2.3.4.

---

The bug pattern seems to want me to use `Duration#dividedBy(Duration)` instead, but these two methods aren't interchangeable.  They answer different questions.

https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/time/Duration.html#dividedBy(long)
https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/time/Duration.html#dividedBy(java.time.Duration)

I notice there is a TODO for inspecting the return types in the bug pattern:
https://github.com/google/error-prone/blob/df4cdefbf351c76098562c2d83eebf4cafaa8eea/core/src/main/java/com/google/errorprone/bugpatterns/time/PreferJavaTimeOverload.java#L345-L348

I tried resolving that TODO by adding another condition...
```java
&& isSameType(input.getReturnType(), calledMethod.getReturnType(), state)
```

...and that made the incorrect warning go away.

**Comment by kluever** [comment]
Thanks - will get this fixed up ASAP

**Comment by kluever** [comment]
Will be mirrored out shortly.
